### PR TITLE
including the actual exception, as part of the 'onShutdown' notification payload

### PR DESCRIPTION
**Pull request by gfaza**
attn @sfgeorge,

abandoned original idea of passing the exception into the notification.  as it turns out, the notification is meant to mimic an Exception, so made more sense to pass the error code instead.  this way we'd simply use the `getCode` method that the Exception interface expects.

### CODE REVIEW

File: src/ErrorHandler.php
Comment (sfgeorge) [inline src/ErrorHandler.php:70]: Thanks for doing this, man.

Could you also update `onError()` such that each of the Base objects that it creates is given the `$code` as the 5th argument to the constructor?

Comment (gfaza) [reply src/ErrorHandler.php:70]: 'tis done

**Review by sfgeorge — APPROVED** [review]
This looks good, thank you mate! 👍 

I suggest we merge PR #2 first to make the test suite happy, just so that we don't get a false-positive and make it look like this PR broke something that it didn't

**Comment by sfgeorge** [comment]
@ajayanty Can you merge this after merging #2?

**Pull request merged**
